scheduler/src/cook/mesos/constraints.clj

```diff
@@ -234,6 +235,19 @@
         (when (< 0 max-expected-runtime)
           (->estimated-completion-constraint expected-end-time host-lifetime-mins))))))
 
+(defn build-launch-max-tasks-constraint
+  "This returns a Fenzo hard constraint that ensures that we don't match more than a given number of tasks per cycle."
+  []
+  (let [enforcing? (ratelimit/enforce? ratelimit/global-job-launch-rate-limiter)
+        max-tasks (ratelimit/get-token-count! ratelimit/global-job-launch-rate-limiter ratelimit/global-job-launch-rate-limiter-key)]
+    (reify com.netflix.fenzo.ConstraintEvaluator
```

Comment:
JobConstraint's API hides the relevant API, not that TaskTrackerState isn't bound to a variable when we do the bridge form JobConstraint to ConstraintEvaluator. Fixing this would require redoing the existing JobConstraint API, expanding it to include the extra state variable and using/ignoring it + JobConstraint.job-constraint-evaluate seems weird. The 3-arg version should probably be called 'job-constraints-evalate-for-rebalancer' to better denote what it means --- or perhaps drop the JobConstraint API layer completely. Not sure how much it buys us above a couple of convenient utility functions. 

(defn fenzoize-job-constraint
  "Makes the JobConstraint 'constraint' Fenzo-compatible."
  [constraint]
  (reify com.netflix.fenzo.ConstraintEvaluator
    (getName [_] (job-constraint-name constraint))
    (^com.netflix.fenzo.ConstraintEvaluator$Result evaluate
      [^com.netflix.fenzo.ConstraintEvaluator _
       ^com.netflix.fenzo.TaskRequest task-request
       ^com.netflix.fenzo.VirtualMachineCurrentState target-vm
       ^com.netflix.fenzo.TaskTrackerState _]